plugin/src/main/kotlin/org/jetbrains/intellij/tasks/InsertPluginsTask.kt

```diff
@@ -0,0 +1,88 @@
+package org.jetbrains.intellij.tasks
+
+import org.gradle.api.internal.ConventionTask
+import org.gradle.api.tasks.*
+import org.jetbrains.intellij.InspectionPlugin
+import org.jetbrains.intellij.extensions.InspectionsExtension
+import org.jetbrains.intellij.versions.IdeaVersion
+import org.jetbrains.intellij.versions.KotlinPluginVersion
+import java.io.File
+
+open class InsertPluginsTask : ConventionTask() {
+
+    @get:Input
+    val ideaVersion: IdeaVersion
+        get() = InspectionPlugin.ideaVersion(extension.ideaVersion)
+
+    @get:Input
+    @get:Optional
+    val kotlinPluginVersion: KotlinPluginVersion?
+        get() = InspectionPlugin.kotlinPluginVersion(extension.kotlinPluginVersion, extension.kotlinPluginLocation)
+
+    @get:InputDirectory
+    val ideaDirectory: File
+        get() = InspectionPlugin.ideaDirectory(ideaVersion)
+
+    @get:InputDirectory
+    @get:Optional
+    val kotlinPluginDirectory: File?
+        get() = kotlinPluginVersion?.let { InspectionPlugin.kotlinPluginDirectory(it) }
+
+    @get:OutputDirectory
+    val destinationIdeaDirectory: File
+        get() = InspectionPlugin.ideaDirectory(ideaVersion, kotlinPluginVersion)
+
+    @Suppress("unused")
+    @TaskAction
+    fun insert() {
+        val idea = ideaDirectory
+        val kotlinPlugin = kotlinPluginDirectory
+        if (isStored(idea, kotlinPlugin)) {
+            logger.info("InspectionPlugin: Kotlin plugin inserting not needed.")
+            return
+        }
+        val destination = destinationIdeaDirectory
+        if (destination.exists())
+            destination.deleteRecursively()
+        project.copy {
+            it.from(idea)
+            it.into(destination)
+        }
+        val pluginsDirectory = InspectionPlugin.pluginsDirectory(destination)
+        val ideaKotlinPluginDirectory = File(pluginsDirectory, "Kotlin")
+        if (kotlinPlugin == null) {
+            logger.info("InspectionPlugin: Using kotlin plugin inherit from idea. No inserting needed.")
+            store(idea, kotlinPlugin)
+            if (ideaKotlinPluginDirectory.exists()) return
+            logger.error("InspectionPlugin: Kotlin plugin not found. Try specify kotlin plugin version.")
+            throw IllegalArgumentException("InspectionPlugin: Kotlin plugin not found. Try specify kotlin plugin version.")
+        }
+        val kotlinPluginFile = File(kotlinPlugin, "Kotlin/lib/kotlin-plugin.jar")
+        val ideaKotlinPluginFile = File(ideaKotlinPluginDirectory, "Kotlin/lib/kotlin-plugin.jar")
+        if (ideaKotlinPluginFile.exists()) ideaKotlinPluginFile.delete()
+        kotlinPluginFile.compareTo(ideaKotlinPluginFile)
```

Comment:
Please test this on `org.jetbrains.kotlin.idea.inspections.CanSealedSubClassBeObjectInspection`, introduced in Kotlin 1.2.50